### PR TITLE
Fix some of clippy warnings

### PR DESCRIPTION
**Pull request by chansuke**
Fix some of clippy warnings

### CODE REVIEW

File: tests/socket.rs [hunk 1]
```diff
@@ -623,7 +623,7 @@ fn recv_vectored_truncated() {
         .recv_vectored(&mut [MaybeUninitSlice::new(&mut buffer)])
         .unwrap();
     assert_eq!(received, 24);
-    assert_eq!(flags.is_truncated(), true);
+    assert!(flags.is_truncated(), "{}", true);
```
Comment (Thomasdezeeuw) [inline tests/socket.rs:626]: This prints true on failure, which doesn't make a whole lot of sense.

Comment (Thomasdezeeuw) [reply tests/socket.rs:626]: In fact could you revert all of these assert changes, I prefer the left/right message you get with `assert_eq`.

Comment (Darksonn) [reply tests/socket.rs:626]: Yeah, seems like we should just disable that clippy warning on this particular file.

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
Thanks @chansuke.

**Pull request merged**